src/challenges/unlock_manager.cpp

```diff
@@ -257,4 +273,65 @@ void UnlockManager::findWhatWasUnlocked(int points_before, int points_now,
             }
         }
     }
-}
+} // findWhatWasUnlocked
+
+void UnlockManager::unlockByPoints(int points, const StoryModeStatus* status)
+{
+    int easy_challenge = CHALLENGE_POINTS[RaceManager::DIFFICULTY_MEDIUM];
+    int medium_challenge = CHALLENGE_POINTS[RaceManager::DIFFICULTY_MEDIUM];
+    int hard_challenge = CHALLENGE_POINTS[RaceManager::DIFFICULTY_HARD];
+    int best_challenge = CHALLENGE_POINTS[RaceManager::DIFFICULTY_BEST];
+    std::string challenge_id;
+
+    // Unlock SuperTux difficulty
+    // We can't test if it is already unlocked because m_locked_features is undefined
+    // when this function is called
+
+
+    int supertux_floor = hard_challenge*(10+4*GP_FACTOR)+medium_challenge*(11);
```

Comment:
Challenges based on points face a major UI hurdle, the player is never told how many points they need to unlock that